lib/Onyx.js

```diff
@@ -1102,7 +1109,11 @@ function clear(keysToPreserve = []) {
                 notifyCollectionSubscribersOnNextTick(key, value);
             });
 
-            return Storage.multiSet(keyValuesToReset);
+            // Call clear() and make sure that the default key/values and the key/values from the parameter
+            // are preserved in storage. This makes sure to always leave storage in a state that contains
+            // all the default values and any additional values that we want to remain after the database is cleared.
+            return Storage.clear()
+                .then(() => Storage.multiSet(defaultKeyValuePairs.concat(keyValuesToPreserve)));
```

Comment:
Why do we use `.concat` instead of spreading (like we do with props)?

```js
Storage.multiSet([...defaultKeyValuePairs, ...keyValuesToPreserve])
```

both `.concat` and spreading result in a new array

Reply:
I try to stay away from using the spread operator on code that needs to be super performant. Babel will transpile to something that's pretty slow. It's probably not really necessary here, but it's the habit I've been in.

Reply:
This might be `true` for objects, if some of the keys are destructed and the rest are `...rest`, but for Arrays, babel works like this 

Code
```js
const arr3 = [...arr1, ...arr2];
```

Transpiled
```js
var arr3 = [].concat(arr1, arr2);
```

You can check that out in the live https://babeljs.io/repl though I think babel stopped transpiling the spread operator as it is supported natively by all alive browsers: https://caniuse.com/?search=spread%20operator

I think if we tell webpack/babel to ignore IE11 it would just stop transpiling the spread operator

Reply:
Ah, that would be great to stop transpiling it altogether. We are not interested in supporting IE11. I didn't know that about arrays. I'll go ahead and switch this to a spread operator.

Reply:
> Ah, that would be great to stop transpiling it altogether. We are not interested in supporting IE11. I didn't know that about arrays. I'll go ahead and switch this to a spread operator.

Here's a small PR regarding that: https://github.com/Expensify/react-native-onyx/pull/226